middleware/src/middleware.go

```diff
@@ -493,35 +496,65 @@ func (middleware *Middleware) EnableClearnetIBD(toggleAction rpcmessages.ToggleS
 	return rpcmessages.ErrorResponse{Success: true}
 }
 
-// ShutdownBase returns an ErrorResponse struct in response to a rpcserver request
-// It calls the bbb-cmd.sh script which initializes a shutdown
+// ShutdownBase shuts the Base down.
+// The shutdown is exectuted in a goroutine with a delay of a few seconds.
+// Prior to starting the goroutine the path for the `shutdown` executable is checked.
+// If the executable is found, a ErrorResponse indicating success is returned.
+// Otherwise a ExecutableNotFound Code is returned.
 func (middleware *Middleware) ShutdownBase() rpcmessages.ErrorResponse {
-	log.Println("shutting down the Base via the cmd script")
-	out, err := middleware.runBBBCmdScript([]string{"base", "shutdown"})
-	if err != nil {
-		errorCode := handleBBBScriptErrorCode(out, err, nil)
-		return rpcmessages.ErrorResponse{
-			Success: false,
-			Message: strings.Join(out, "\n"),
-			Code:    errorCode,
+	const shutdownDelay time.Duration = 5 * time.Second
+	log.Printf("Shutting down the Base in %s\n", shutdownDelay)
+
+	if !middleware.isMockmode {
```

Comment:
I think I'd prefer 
```
if middleware.isMockmode {
    return rpcmessages.ErrorResponse{Success: true}
}
```
and avoid having the entire block in an if statement. Also more consistent with cmd and config sh scripts which also just check for MOCKMODE first and then execute the rest

Reply:
This results in having it as:

```go
func (middleware *Middleware) ShutdownBase() rpcmessages.ErrorResponse {
  ...
  if middleware.isMockmode {
      return rpcmessages.ErrorResponse{Success: true}
  } 
  ...
  return rpcmessages.ErrorResponse{Success: true}
}
```

Not super clean, but better. I'll change it.